test/mount_test/mount_test.go

```diff
@@ -0,0 +1,149 @@
+// +build !unittest
+
+package mount_test
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+)
+
+var blobfuseBinary string = "blobfuse2"
+var mntDir string = "mntdir"
+var configFile string
+
+type mountSuite struct {
+	suite.Suite
+}
+
+func remountCheck(suite *mountSuite) {
+	mountCmd := exec.Command(blobfuseBinary, "mount", mntDir, "--config-file="+configFile)
+	cliOut, err := mountCmd.Output()
+	fmt.Println(string(cliOut))
+	suite.NotEqual(0, len(cliOut))
+	suite.NotEqual(nil, err)
+	suite.Contains(string(cliOut), "directory is already mounted")
+}
+
+// list blobfuse mounted directories
+func listBlobfuseMounts(suite *mountSuite) []byte {
+	mntListCmd := exec.Command(blobfuseBinary, "mount", "list")
+	cliOut, err := mntListCmd.Output()
+	fmt.Println(string(cliOut))
+	suite.Equal(nil, err)
+	return cliOut
+}
+
+// mount command test along with remount on the same path
+func (suite *mountSuite) TestMountCmd() {
+	// run mount command
+	mountCmd := exec.Command(blobfuseBinary, "mount", mntDir, "--config-file="+configFile)
+	cliOut, err := mountCmd.Output()
+	fmt.Println(string(cliOut))
+	suite.Equal(0, len(cliOut))
+	suite.Equal(nil, err)
+
+	// wait for mount
+	time.Sleep(10 * time.Second)
+
+	// validate mount
+	cliOut = listBlobfuseMounts(suite)
+	suite.NotEqual(0, len(cliOut))
+	suite.Contains(string(cliOut), mntDir)
+
+	remountCheck(suite)
+
+	// unmount
+	unmountCmd := exec.Command(blobfuseBinary, "unmount", "all")
+	cliOut, err = unmountCmd.Output()
+	fmt.Println(string(cliOut))
+	suite.NotEqual(0, len(cliOut))
+	suite.Equal(nil, err)
+	suite.Contains(string(cliOut), mntDir)
+
+	// wait after unmount
+	time.Sleep(5 * time.Second)
+
+	// validate unmount
+	cliOut = listBlobfuseMounts(suite)
+	suite.Equal(0, len(cliOut))
```

Comment:
Shouldn't we call unmount also in each test case.

Reply:
Added the unmount in each test case